openverse_catalog/dags/maintenance/add_license_url.py

```diff
@@ -0,0 +1,162 @@
+        logger.info(f"Processing {license_name} {license_version}, {license_items}.")
+        license_url = f"{base_url}{path}/"
+
+        select_query = dedent(
+            f"""
+            SELECT identifier FROM image
+            WHERE (
+            meta_data is NULL AND license = '{license_name}'
+            AND license_version = '{license_version}')
+            """
+        )
+        result = postgres.get_records(select_query)
+
+        if not result:
+            logger.info(f"No records to update with {license_url}.")
+            continue
+        logger.info(f"{len(result)} records to update with {license_url}.")
+        license_url_col = {"license_url": license_url}
+        update_license_url_query = dedent(
+            f"""
+            UPDATE image
+            SET meta_data = {Json(license_url_col)}
+            WHERE identifier IN ({','.join([f"'{r[0]}'" for r in result])});
+            """
+        )
+
+        updated_count = postgres.run(
+            update_license_url_query, autocommit=True, handler=RETURN_ROW_COUNT
+        )
+        logger.info(f"{updated_count} records updated with {license_url}.")
+        total_counts[license_url] = updated_count
+        total_count += updated_count
+
+    logger.info(
+        f"{total_count} image records with missing license_url updated.\n"
+        f"Total updated counts: {total_counts}"
+    )
+    return total_counts
+
+
+def final_report(postgres_conn_id: str, item_count):
+    null_meta_data_count = get_null_counts(postgres_conn_id)
+
+    message = f"""
+Added license_url to *{item_count}* items`
+Now, there are {null_meta_data_count} records with NULL meta_data left.
+"""
+    send_message(
+        message,
+        username="Airflow DAG Data Normalization - license_url",
+        dag_id=DAG_ID,
+    )
+
+    logger.info(message)
+
+
+dag = DAG(
+    dag_id=DAG_ID,
+    default_args={
+        **DAG_DEFAULT_ARGS,
+        "schedule_interval": None,
+        "retries": 0,
+    },
+    schedule_interval=None,
+    catchup=False,
+    # Use the docstring at the top of the file as md docs in the UI
+    doc_md=__doc__,
+    tags=["data_normalization"],
+)
+
+with dag:
+    get_statistics = BranchPythonOperator(
+        task_id="get_stats",
+        python_callable=get_statistics,
+        op_kwargs={"postgres_conn_id": POSTGRES_CONN_ID},
+    )
+    update_license_url = PythonOperator(
+        task_id=UPDATE_LICENSE_URL,
+        python_callable=update_license_url,
+        op_kwargs={"postgres_conn_id": POSTGRES_CONN_ID},
+    )
+    final_report = PythonOperator(
+        task_id=FINAL_REPORT,
+        python_callable=final_report,
+        op_kwargs={
+            "postgres_conn_id": POSTGRES_CONN_ID,
+            "item_count": XCOM_PULL_TEMPLATE.format(
+                update_license_url.task_id, "return_value"
+            ),
+        },
+    )
+
+    get_statistics >> [update_license_url, final_report]
+    update_license_url >> final_report
```

Comment:
I think this could be written more simply as `get_statistics >> update_license_url >> final_report`

Reply:
If we simplify it, would it skip the `update_license_url` step if the `get_statistics` says that we don't need to update anything? The intention was to skip the update step if we ever run this on a fully cleaned dataset, but I'm not very good with branching operators :)

Reply:
Ahh, that's right! Do you mind adding a comment to that effect Olga? You're totally right, the current setup is required for the branching operator, but I could see making that change in the future and accidentally breaking something 😅

Reply:
You're totally correct, I missed that there was a branching operator! 😅